src/auxiliary/caches.jl

```diff
@@ -0,0 +1,148 @@
+const GLOBAL_CACHES = Any[]
+function empty_globalcaches!()
+    foreach(empty!, GLOBAL_CACHES)
+    return nothing
+end
```

Comment:
Is there a way to have some info display function as well here? Related to this, storing the `GLOBAL_CACHES` as a vector does lose some of the information about which cache it is, it might be reasonable to use a `Dict` or `IdDict` instead to keep the name around, or store the name together with the cache